server/controllers/finance/debtorGroups.js

```diff
@@ -0,0 +1,176 @@
+/**
+* The Debtor Groups Controllers
+*
+* @module finance/debtorGroups
+*
+* @desc This module is responsible for handling all crud operations relatives
+* to debtor groups, and relatives functions
+*
+* @required q
+* @required lib/db
+* @required lib/guid
+*
+* @todo implements the creation of a Debtor Group
+*/
+
+'use strict';
+
+var q  = require('q'),
+    db = require('../../lib/db'),
+    guid = require('../../lib/guid');
+
+/** Get debtor group details */
+exports.detail = getDetail;
+
+/** Get the list of debtor group */
+exports.list = getList;
+
+/** [HTTP API ENDPOINT] Get debtor groups invoices list */
+exports.fetchInvoices = fetchInvoices;
+
+/** [SERVER SIDE FUNCTION] Get debtor groups invoices list */
+exports.getInvoices = getInvoices;
+
+/**
+* GET /debtor_groups/:uuid
+*
+* @function getDetail
+*
+* @desc This function is responsible for retrieving details of a debtor group
+*/
+function getDetail(req, res, next) {
+  var uuid = req.params.uuid;
+
+  var debtorDetailsQuery =
+    'SELECT uuid, name, account_id, location_id, phone, email, note, locked ' +
+      'max_credit, is_convention, price_list_uuid ' +
+    'FROM debitor_group ' +
+    'WHERE uuid = ?';
+
+  db.exec(debtorDetailsQuery, [uuid])
+  .then(function (rows) {
+    if (!rows.length) { return next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+
+}
+
+/**
+* GET /debtor_groups/
+* GET /debtor_groups/?with_locked={1|0}
+*
+* @function getList
+*
+* @desc This function is responsible for retrieving details of a debtor group
+*/
+function getList(req, res, next) {
+  var query;
+
+  query = 'SELECT uuid, name, locked, account_id FROM debitor_group ';
+  query += (req.query.with_locked === '1') ? '' : 'WHERE locked = 0';
+
+  db.exec(query)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* GET /debtor_groups/{:uuid}/invoices
+* GET /debtor_groups/{:uuid}/invoices?balanced=1
+*
+* @function fetchInvoices
+*
+* @desc This function is responsible for getting all invoices of a specified debtor group
+*/
+function fetchInvoices(req, res, next) {
+  getInvoices(req.params.uuid, req.query.balanced)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(function (err) {
+    next(err);
+  })
+  .done();
+}
+
+/**
+* Get Debtor group invoices
+*
+* @function getInvoices
+*
+* @param {number} uuid The debtor group uuid
+*
+* @return {array} An promise of an array
+*/
+function getInvoices (id, balanced) {
+
+  if (!id) { return q.reject(new Error('ERR_MISSING_REQUIRED_PARAMETERS')); }
+
+  var query =
+    'SELECT `debitor_group`.`account_id` FROM `debitor_group` ' +
+    'WHERE `debitor_group`.`uuid`= ?';
+
+  return db.exec(query, [id])
+  .then(function (rows) {
+    if (!rows.length) { return q.resolve([]); }
```

Comment:
I think you don't need `q.resolve()` here.  Because you are in a promise chain, a simple `return [[;` should work.

For example, this will work no problem:
```js
q(4)
.then(function (number) {
  // in this block, number === 4
  // the q(4) created a promise that was fulfilled immediately
  return {};
})
.then(function (object) {
  // in this block, the first parameter is an object, not the number 4
  // the previous .then() block simply returned an object, which was passed as the first paramter
  // to the next .then() function immediately.
  return 3;
})
.then(function (number) {
  // here, number is 3, since it was returned from the previous block
  // even though we didn't return a promise from the two blocks in the chain, whatever result we return
  // will be automatically passed to the next .then() method.
  return db.exec(sql, params);
})
.then(function (rows) {
  // finally, this block has the rows returned by the `db.exec()`.  Since the previous block returned a promise,
  // the .then() automatically waited for the db.exec() promise to be fulfilled before executing this block of code.
})
.catch(function (error) {
  // this block will never be triggered, unless the db.exec() promise throws an error.
  // there is no other place in the previous promise chain that an error could have been thrown.
});
```